packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -539,28 +668,49 @@ contract OPContractsManager is ISemver {
         return abi.encodeCall(IDelayedWETH.initialize, (_input.roles.opChainProxyAdminOwner, superchainConfig));
     }
 
-    function encodePermissionedDisputeGameConstructor(
-        DeployInput memory _input,
-        DeployOutput memory _output
+    function encodePermissionlessFDGConstructor(IFaultDisputeGame.GameConstructorParams memory _params)
+        internal
+        view
+        virtual
+        returns (bytes memory)
+    {
+        return abi.encode(
```

Comment:
We can make this type safe by using `abi.encodeCall(IFaultDisputeGame.__constructor__, (_params))`

Reply:
I wasn't able to get this to work - I think because it's being passed into a blueprint, we need a raw abi.encode call rather than abi.encodeCall which will include a selector.

Reply:
Yeah, that sounds right about the additional selector not working. 
That said, not using encodeCall, especially when this many args are involved can make things very hard to debug.

It's a big ugly, but what I'd suggest is that we use `abi.encodeCall(IFaultDisputeGame.__constructor__, (_params))` then cut off the first 4 bytes. I'll see if I can dig up an example of something like that being done.

Reply:
```
import { Bytes } from "src/libraries/Bytes.sol";

...

        bytes memory dataWithSelector = abi.encodeCall(IFaultDisputeGame.__constructor__, (_params));

        return Bytes.slice(dataWithSelector, 4, dataWithSelector.length - 4);
```